File: Problems/Two Sum.py
class Solution:
    def isPairSum(self, nums, target):
        left = 0
        right = length(nums) - 1
        while (left < right):
          	pairSum = nums[left] + nums[right]
          	if (pairSum == target):
          	    return {nums[left], nums[right]}
          	elif (pairSum < target):
          	    left += 1
          	else:
          	    right -= 1
          
        return {-1, -1}

File: Problems/Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> pairIdx = new HashMap<>(); // create map to store values

        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            if (pairIdx.containsKey(target - num)) { // subtract each number from target and see if value is present in map
                return new int[] { i, pairIdx.get(target - num) };
            }
            pairIdx.put(num, i);
        }

        return new int[] {};        
    }
}
